doc/contributing/building_ruby.md

```diff
@@ -0,0 +1,77 @@
+# Building Ruby
+
+## Quick start guide
+
+1. Install the prerequisite dependencies for building the CRuby interpreter:
+
+    * C compiler
+    * autoconf - 2.67 or later
+    * bison - 2.0 or later
+    * gperf - 3.0.3 or later
+    * ruby - 2.7 or later
+
+2. Install optional, recommended dependencies:
+
+    * OpenSSL/LibreSSL
+    * readline/editline(libedit
+    * zlib
+    * libffi
+    * libyaml
+    * libexecinfo (FreeBSD)
+    * NDBM/QDBM
+    * GDBM
+
+3. Checkout the CRuby source code:
+
+    ```
+    git clone https://github.com/ruby/ruby.git ruby-master
```

Comment:
What's the point to use the specific directory `ruby-master` for the git repository? The directory is not only for master branch, but also all the branches. I think just `git clone https://github.com/ruby/ruby.git` might be better.

For example, in the following case, the `ruby-master` is not for the master branch any more.

```
$ git clone https://github.com/ruby/ruby.git ruby-master
$ cd ruby-master
$ git checkout v3_1_2
```

Reply:
> What's the point to use the specific directory `ruby-master` for the git repository? The directory is not only for master branch, but also all the branches. I think just `git clone https://github.com/ruby/ruby.git` might be better.

Yes. If somebody has several checkouts, they will probably have their own system of naming them. And if we every decide to change the name of the main branch from 'master' to something else, this makes it one less place we'd have to change.